multicluster/cmd/multicluster-controller/leader.go

```
@@ -0,0 +1,94 @@
+		Short: "Run the controller in leader cluster",
+		Long:  "Run the Antrea Multi-Cluster controller for leader cluster",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := opts.complete(args); err != nil {
+				klog.Fatalf("Failed to complete: %v", err)
+			}
+			if err := runLeader(opts); err != nil {
+				klog.Fatalf("Error running controller: %v", err)
+			}
+		},
+	}
+
+	return leaderCmd
+}
+
+func runLeader(o *Options) error {
+	// on the leader we want the reconciler to run for a given namspace instead of cluster scope
+	o.options.Namespace = env.GetPodNamespace()
+
+	mgr, err := setupManagerAndCertController(o, true)
+	if err != nil {
+		return err
+	}
+
+	clusterSetReconciler := &multiclustercontrollers.LeaderClusterSetReconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}
+	if err = clusterSetReconciler.SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "ClusterSet")
+		return fmt.Errorf("unable to create ClusterSet controller, err: %v", err)
+	}
+
+	if err = (&multiclustercontrollers.MemberClusterAnnounceReconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "MemberClusterAnnounce")
+		return fmt.Errorf("unable to create MemberClusterAnnounce controller, err: %v", err)
+	}
+	if err = (&multiclusterv1alpha1.MemberClusterAnnounce{}).SetupWebhookWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create webhook", "webhook", "MemberClusterAnnounce")
+		return fmt.Errorf("unable to create MemberClusterAnnounce webhook, err: %v", err)
+	}
+
+	resExportReconciler := &multiclustercontrollers.ResourceExportReconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme()}
+	if err = resExportReconciler.SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "ResourceExport")
+		os.Exit(1)
```

here we should return an error and leave main.go to handle exit?